ndarray/src/main/java/org/tensorflow/ndarray/impl/dense/AbstractDenseNdArray.java

```diff
@@ -43,18 +44,29 @@ public NdArraySequence<U> elements(int dimensionIdx) {
     DimensionalSpace elemDims = dimensions().from(dimensionIdx + 1);
     try {
       DataBufferWindow<? extends DataBuffer<T>> elemWindow = buffer().window(elemDims.physicalSize());
-      U element = instantiate(elemWindow.buffer(), elemDims);
+      U element = instantiateView(elemWindow.buffer(), elemDims);
       return new FastElementSequence(this, dimensionIdx, element, elemWindow);
     } catch (UnsupportedOperationException e) {
       // If buffer windows are not supported, fallback to slicing (and slower) sequence
       return new SlicingElementSequence<>(this, dimensionIdx, elemDims);
     }
   }
 
+  @Override
+  public U withShape(Shape shape) {
+    if (shape == this.shape()) {
```

Comment:
Shouldn't this be `this.shape.equals(shape)` rather than `==`?

Reply:
Yep... I might have rushed that PR a bit too quickly, thanks for spotting this

Reply:
I just changed the unit test to pass a new (and equal) shape. The test was passing the same instance, hence why it was working I suppose..